truss/cli/cli.py

```diff
@@ -237,6 +247,147 @@ def watch(
+        if status == ACTIVE_STATUS:
+            spinner_name = "active"
+        elif status in DEPLOYING_STATUSES:
+            spinner_name = "deploying"
+        else:
+            spinner_name = "failed"
+        spinner = rich.spinner.Spinner(spinner_name, text=status)
+        table.add_row(spinner, name, logs_url)
+        statuses.append(status)
+    return table, statuses
+
+
+@chain.command()  # type: ignore
+@click.argument("source", type=Path, required=True)
+@click.argument("entrypoint", type=str, required=True)
+@click.option(
+    "--name",
+    type=str,
+    required=False,
+    help="Name of the chain to be deployed, if not given, the entrypoint name is used.",
+)
+@click.option(
+    "--publish",
+    type=bool,
+    default=False,
+    is_flag=True,
+    help="Create chainlets as published deployments.",
+)
+@click.option(
+    "--promote",
+    type=bool,
+    default=False,
+    is_flag=True,
+    help="Replace production chainlets with newly deployed chainlets.",
+)
+@click.option(
+    "--wait",
+    type=bool,
+    default=True,
+    is_flag=True,
+    help="Wait until all chainlets are ready (or deployment failed).",
+)
+@click.option(
+    "--dryrun",
+    type=bool,
+    default=False,
+    is_flag=True,
+    help="Produces only generated files, but doesn't deploy anything.",
+)
+@error_handling
+def deploy(
+    source: Path,
+    entrypoint: str,
+    name: Optional[str],
+    publish: bool,
+    promote: bool,
+    wait: bool,
+    dryrun: bool,
+) -> None:
+    """
+    Deploys a chain remotely.
+
+    SOURCE: Path to a python file that contains the entrypoint chainlet.
+
+    ENTRYPOINT: Class name of the entrypoint chainlet in source file.
+    """
+    chain_name = name or entrypoint
+    entrypoint_cls = framework.import_target(source, entrypoint)
+    options = chains_deploy.DeploymentOptionsBaseten.create(
+        chain_name=chain_name,
+        promote=promote,
+        publish=publish,
+        only_generate_trusses=dryrun,
+    )
+    service = chains_deploy.deploy_remotely(entrypoint_cls, options)
+
+    console.print("\n")
+    if dryrun:
+        return
+
+    run_help_msg = (
+        f"curl -X POST '{service.run_url}' \\\n"
+        '    -H "Authorization: Api-Key $BASETEN_API_KEY" \\\n'
+        "    -d '<JSON_INPUT>'"
+    )
+
+    table, statuses = _create_chains_table(service)
+    if wait:
+        num_services = len(statuses)
+        success = False
+        with rich.live.Live(table, console=console, refresh_per_second=4) as live:
+            while True:
+                table, statuses = _create_chains_table(service)
+                live.update(table)
+                num_active = sum(s == ACTIVE_STATUS for s in statuses)
+                num_deploying = sum(s in DEPLOYING_STATUSES for s in statuses)
+                if num_active == num_services:
+                    success = True
+                    break
+                elif num_services - num_active - num_deploying:
```

Comment:
this condition means that some of them failed, is there any way we could make that clearer? ie:  num_failed > 0 ?